sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -0,0 +1,207 @@
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name]
+    )
+    def test_any_user_should_not_have_superuser_property_except_superuser(self, user_type):
+        self.assertFalse(Command._set_user_factory_parameters(user_type).get("is_superuser"))
+
+    def test_superuser_should_have_superuser_property(self):
+        self.assertTrue(Command._set_user_factory_parameters(self.superuser).get("is_superuser"))
+
+    @parameterized.expand([CustomUser.UserType.ADMIN.name, superuser])
+    def test_admin_and_superuser_should_have_user_type_coded_as_admin(self, user_type):
+        self.assertEqual(
+            Command._set_user_factory_parameters(user_type).get("user_type"), CustomUser.UserType.ADMIN.name
+        )
+
+    @parameterized.expand([CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name])
+    def test_employee_and_manager_should_have_user_type_coded_as_based_of_user_type_name(self, user_type):
+        self.assertEqual(Command._set_user_factory_parameters(user_type).get("user_type"), user_type)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 5),
+            (CustomUser.UserType.MANAGER.name, 3),
+        ]
+    )
+    def test_result_of_create_user_function_should_be_specified_number_of_users_in_database(
+        self, user_type, number_of_users_to_create
+    ):
+        Command().create_user(user_type, number_of_users_to_create)
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_to_create)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 4, 3),
+            (CustomUser.UserType.MANAGER.name, 2, 1),
+        ]
+    )
+    def test_create_user_function_should_not_create_users_if_number_in_database_is_greater_than_or_equal_to_specified(
+        self, user_type, number_of_users_in_database, number_of_users_to_create
+    ):
+        Command().create_user(user_type, number_of_users_in_database)
+        Command().create_user(user_type, number_of_users_to_create)
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_in_database)
+
+
+class CreateProjectMethodsTests(TestCase):
+    def setUp(self):
+        self.options = {ProjectType.SUSPENDED.name: 4, ProjectType.ACTIVE.name: 5, ProjectType.COMPLETED.name: 7}
+
+        for _project_number in range(2):
+            ProjectFactory(suspended=True)
+
+        for _project_number in range(2):
+            ProjectFactory()
+
+        for _project_number in range(2):
+            ProjectFactory(stop_date=timezone.now() - relativedelta(days=1))
+
+        self.number_of_suspended_projects_to_create = (
+            self.options.get(ProjectType.SUSPENDED.name) - Project.objects.filter_suspended().count()
```

Comment:
To avoid query after database results here you can handle differently creating test projects. Instead of creating about Projects without handling return instance from `ProjectFactory` handle it inside list and then just check length of it. That's how you can simplify this setUp part.
For example:
```python
def _generate_test_projects(number_of_projects, project_type):
    project_list = []
    for _ in range(number_of_projects):
        if project_type == ProjectType.SUSPENDED.name:
            project = ProjectFactory(suspended=True)
        elif ....

        project_list.append(project)
    return project_list

self.suspended_projects = self._generate_test_projects(number_of_projects=3, project_type=ProjectType.SUSPENDED.name)

self.suspended_projects_to_create = self.options.[ProjectType.SUSPENDED.name] - len(self.suspended_projects)

```

Reply:
Done